apiserver/resources_unit.go

```diff
@@ -63,6 +63,12 @@ func (h *UnitResourcesHandler) ServeHTTP(resp http.ResponseWriter, req *http.Req
 			logger.Errorf("unable to complete stream for resource: %v", err)
 			return
 		}
+
+		// Mark the downloaded resource as in use on the unit.
+		err = opener.SetResource(req.Context(), name)
```

Comment:
It's as if you intended to use the unit name for this, but added the anonymous setter func.